definitions/features/foreman_proxy.rb

```diff
@@ -1,21 +1,58 @@
 class Features::ForemanProxy < ForemanMaintain::Feature
   metadata do
     label :foreman_proxy
+
+    confine do
+      file_exists?('/etc/dhcp/dhcpd.conf') &&
+        ForemanMaintain::Utils::SmartProxySSL.instance.ssl_certs_exists?
+    end
   end
 
-  attr_reader :dhcpd_conf_file, :cert_path
+  attr_reader :dhcpd_conf_file, :cert_path, :base_url,
+              :default_ssl_cert_options
 
   def initialize
     @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
     @cert_path = ForemanMaintain.config.foreman_proxy_cert_path
+    @base_url = construct_foreman_proxy_url
+    @default_ssl_cert_options = load_default_ssl_cert_options
   end
 
   def valid_dhcp_configs?
     dhcp_req_pass? && !syntax_error_exists?
   end
 
+  def list_reservations(subnet)
+    dhcp_api_client.list(subnet)
+  end
+
+  def add_reservation(subnet, ip, mac_address, name)
+    dhcp_api_client.add(subnet, ip, mac_address, name)
+  end
+
+  def remove_reservation(subnet, ip)
+    dhcp_api_client.delete(subnet, ip)
+  end
+
   private
 
+  def construct_foreman_proxy_url
+    proxy_host = ForemanMaintain.config.foreman_proxy_fqdn
+    proxy_host = hostname if proxy_host.empty?
+    URI::HTTPS.build(
+      :host => proxy_host,
+      :port => ForemanMaintain.config.foreman_proxy_port.to_i
+    ).to_s
+  end
+
+  def dhcp_api_client
+    ForemanMaintain::Utils::DhcpReservation.new("#{base_url}/dhcp", default_ssl_cert_options)
+  end
+
+  def load_default_ssl_cert_options
+    ForemanMaintain::Utils::SmartProxySSL.instance.openssl_certs_hash
+  end
+
   def dhcp_curl_cmd
     "curl -w '\n%{http_code}' -slient -ks --cert #{cert_path}/client_cert.pem \
       --key #{cert_path}/client_key.pem \
```

Comment:
@swapab ,

Yes, I have noticed this before but I don't want to mix these changes in this PR.

Not only `port` but now we have introduced `rest-client` we don't even need `curl` here so that we can handle response in better way.
Once this PR get merged, I will modify this code as per new changes.